compiler/src/iree/compiler/Dialect/Stream/Transforms/ScheduleAllocation.cpp

```diff
@@ -1386,14 +1393,24 @@ static SmallVector<ConstantAllocation> extractConstantsWithLifetime(
     }
 
     // Append to affinity bucket.
-    constantOps[allocationAffinity].push_back(constantOp);
-  }
-  if (constantOps.empty()) {
-    return {};
+    if (resultValue) {
+      globalConstantOps[allocationAffinity].push_back(constantOp);
+    } else {
+      localConstantOps[allocationAffinity].push_back(constantOp);
+    }
```

Comment:
I must miss something, because I don't follow why it has different behavior.

So this changes breaks the `constantOps` into two buckets. One is `globalConstantOps` and the other is `localConstantOps`. It implies that `globalConstantOps + localConstantOps = constantOps`.

We remove the empty check, which should be NFC to the cases.

The below two loops have the same implementation, which iterates two containers and push them to `constantAllocations`. The only difference to me is order. I don't follow how it fixes the issue.

Reply:
they are independent SSA values, so one can go out of scope and be deallocated before the other. the point is that they are the same thing just with different lifetimes.

before:
```
%constants = load_all_constants (global + local)
execute %constants  // uses all (global + local)
// local are now unused and can be deallocated, but can't be because the %constants containing them is stored to a global
util.global.store %constants
```

now:
```
%local_constants = load_some_constants
%global_constants = load_some_constants
execute %local_constants, %global_constants
// %local_constants can now be deallocated as execute was the last use
// %global_constants remains live
util.global.store %global_constants
```